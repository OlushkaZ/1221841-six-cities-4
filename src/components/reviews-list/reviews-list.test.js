import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const props = {
  reviews: [
    {
      comment: `Cool`,
      date: `12 April`,
      rating: 5,
      id: 124,
      user: {
        avatar: `img/avatar-angelina.jpg`,
        name: `Peter`,
        isPro: false,
        id: 12,
      }
    },
    {
      comment: `beautiful`,
      date: `12 April`,
      rating: 5,
      id: 12334,
      user: {
        avatar: `img/avatar-angelina.jpg`,
        name: `Peter`,
        isPro: true,
        id: 1442,
      },
    }
  ],
};

const mockStore = configureStore([]);

it(`ReviewsList Render`, () => {
  const store = mockStore({
    USER: {
      authorizationStatus: `NO_AUTH`
    },
  });

  const tree = renderer
  .create(
      <Provider store={store}>
        <ReviewsList {...props} />
      </Provider>
  )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
