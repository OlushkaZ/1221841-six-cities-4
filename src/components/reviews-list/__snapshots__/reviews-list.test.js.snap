// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList Render 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper "
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Peter
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Cool
        </p>
        <time
          className="reviews__time"
          dateTime=""
        >
          April 12
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper property__avatar-wrapper property__avatar-wrapper--pro"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Peter
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          beautiful
        </p>
        <time
          className="reviews__time"
          dateTime=""
        >
          April 12
        </time>
      </div>
    </li>
  </ul>
</section>
`;
